lib/src/git_backend.rs

```diff
@@ -1687,6 +1704,68 @@ mod tests {
         assert_eq!(actual_commit2, commit2);
     }
 
+    #[test]
+    fn write_signed_commit() {
+        let settings = user_settings();
+        let temp_dir = testutils::new_temp_dir();
+        let backend = GitBackend::init_internal(&settings, temp_dir.path()).unwrap();
+
+        let commit = Commit {
+            parents: vec![backend.root_commit_id().clone()],
+            predecessors: vec![],
+            root_tree: MergedTreeId::Legacy(backend.empty_tree_id().clone()),
+            change_id: ChangeId::new(vec![]),
+            description: "initial".to_string(),
+            author: create_signature(),
+            committer: create_signature(),
+            secure_sig: None,
+        };
+
+        let signer = Box::new(|data: &_| {
+            let mut hasher = DefaultHasher::new();
```

Comment:
I don't think `DefaultHasher` is stable across rust versions. Maybe we can use blake2b instead?

Reply:
Oh, in my mind I was equating it to java hashCode, that makes sense